app/models/concerns/authorization_extensions/volumetrie.rb

```diff
@@ -0,0 +1,30 @@
+module AuthorizationExtensions::Volumetrie
+  extend ActiveSupport::Concern
+
+  included do
+    add_attribute :volumetrie_appels_par_minute
+    add_attribute :volumetrie_justification
+
+    validates :volumetrie_appels_par_minute,
+      presence: true,
+      if: -> { need_complete_validation?(:volumetrie) }
+
+    validates :volumetrie_justification,
+      presence: true,
+      if: -> { need_complete_validation?(:volumetrie) && volumetrie_is_not_the_smallest }
+  end
+
+  def volumetrie_is_not_the_smallest
+    volumetrie_appels_par_minute.to_i > default_volumetrie
+  end
+
+  def default_volumetrie
+    available_volumetries.values.min
+  end
+
+  def available_volumetries
+    raise "Must declare a constant VOLUMETRIES in the model #{self.class}, for example {'20 appels / minute' => 20}" unless self.class.const_defined?(:VOLUMETRIES)
+
+    self.class::VOLUMETRIES
```

Comment:
```suggestion
    self.class::VOLUMETRIES
    rescue NameError
      raise "Must declare a constant VOLUMETRIES in the model #{self.class}, for example {'20 appels / minute' => 20}"
    end
```

C'est plus ruby way de faire de cette manière